examples/v2/chat2.nim

```diff
@@ -288,6 +294,8 @@ proc processInput(rfd: AsyncFD, rng: ref BrHmacDrbgContext) {.async.} =
         if decodedPayload.isOK():
           let payload = string.fromBytes(decodedPayload.get().payload)
           echo &"{payload}"
+          chat.prompt = false
```

Comment:
The switching between true and false outside of the proc may get confusing in the future, not sure what is the exact logic, but is it possible to do it inside the `showChatPrompt`?